frontend/pages/views/MapPageView.js

```diff
@@ -198,30 +204,40 @@ class MapPage extends React.Component {
         } catch (e) {
             console.log(e);
         }
+
+        this.updateViewport();
+        window.addEventListener("resize", this.updateViewport);
```

Comment:
This is great logic. The only thing that's missing that will speed things up is a debouncer (a function that slows down how often the inner function, `this.updateViewport` is called). Let's add that to a future/following TODO